app/javascript/gobierto_data/webapp/components/sidebar/SidebarDatasets.vue

```diff
@@ -95,17 +107,17 @@ export default {
   },
   methods: {
     orderDatasets() {
-      const sortDatasets = this.items
-      const allDatasets = sortDatasets.sort((a, b) => a.attributes.name.localeCompare(b.attributes.name));
-      let slug = this.$route.params.id
-      const indexToggle = allDatasets.findIndex(dataset => dataset.attributes.slug == slug)
+      const allDatasets = this.items.sort((a, b) => a.attributes.name.localeCompare(b.attributes.name));
+      let { id } = this.$route.params || {}
+
+      const indexToggle = allDatasets.findIndex(({ attributes: { slug } = {} }) => slug === id)
       this.toggle = indexToggle
       if (this.toggle === -1) {
         this.toggle = 0
-        slug = slug = allDatasets.length ? allDatasets[0].attributes.slug : ''
+        id = allDatasets.length ? allDatasets[0].attributes.slug : ''
```

Comment:
You're mixing unrelated stuff. 

If you need the `id` from $route, you set it in a variable, but don't reassign it, why so? The second `id` is a totally different variable, then you create a new variable for it. These practices are the opposite of code readability.

Then, one more time, destructuration. Why the array Why attributes ought to have a property called `slug`?
```
let anotherId = null;
if (this.toggle.....) {
    ...other stuff

    // destructuration prop by prop, fallbacking them
    const [destructuratedVar: { attributes: { slug } = {} } = {} ] = allDatasets || []
    anotherId = slug

   // same in one line (more complex syntax)
   ([destructuratedVar: { attributes: { slug: anotherId } = {} } = {} ] = allDatasets || [])
}
```